.github/workflows/build-without-push-docker.yml

```diff
@@ -0,0 +1,42 @@
+name: Docker Build without Push
+
+on:
+  pull_request:
+    branches: [main]
+    types:
+    - opened
+    - ready_for_review
+    - reopened
+    - synchronize
+
+
+jobs:
+  docker-build-without-push-frontend:
+    uses: ./.github/workflows/build-docker.yml
```

Comment:
I think it would have been nice to use a matrix here

Reply:
I did consider matrices when I looked at the options. 
I didn't end up using them because we have combinations of configs - 3 dockerfiles each with their own tag labels. WIth a matrix, you end up with the cartesian product of the configs so it didn't quite fit in with our use case.

Reply:
What if you did something like

```yaml
matrix:
  build:
    - name: rag-adminwebapp
      dockerfile: docker/Admin.Dockerfile
    - name: rag-backend
      dockerfile: docker/Backend.Dockerfile
    - name: rag-webapp
      dockerfile: docker/Frontend.Dockerfile
```

That would only create 3 combinations?